src/providers/scicat-provider/dao/mongo-dao.ts

```diff
@@ -50,68 +58,43 @@ export class MongoConnector {
    * @param parameters
    * @returns {Promise<any>}
    */
-  public recordsQuery(parameters: any, filter: MongoClient.filter): Promise<any> {
+  public async recordsQuery(parameters: any, filter: any): Promise<any> {
     if (!this.db) {
       reject("no db connection");
     }
     let Publication = this.db.collection(this.collectionName);
-    return new Promise((resolve: any, reject: any) => {
-      Publication.find(filter).toArray(function(err, items) {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(items);
-        }
-      });
-    });
+    return Publication.find(filter).toArray() ;
   }
 
   /**
    * Responds to OAI ListIdentifiers requests.
    * @param parameters
    * @returns {Promise<any>}
    */
-  public identifiersQuery(parameters: any, filter: MongoClient.filter): Promise<any> {
+  public async identifiersQuery(parameters: any, filter: any): Promise<any> {
     if (!this.db) {
       reject("no db connection");
     }
     let Publication = this.db.collection(this.collectionName);
-    return new Promise((resolve: any, reject: any) => {
-      // need to add relevant date to projection
-      Publication.find(filter, { _id: 1 }).toArray(function(err, items) {
-        if (err) {
-          reject(err);
-        } else {
-          resolve(items);
-        }
-      });
-    });
+    return await Publication.find(filter /* , { _id: 1 } */).toArray() ;
```

Comment:
I think the `{_id: 1}` was done to only return the _id from the query and not the full document. Is this commented on purpose?

Reply:
Fixed. Just forgot to remove it.